### ISSUE TITLE
Symfony recipes are disabled when flex set as dev dependency

### ISSUE THREAD

**Issue by jlekowski** [symfony/flex#414]
Since `symfony/flex 1.1` (https://github.com/symfony/flex/pull/409) when I run `composer install` I get `Symfony recipes are disabled: "symfony/flex" is not in the "require" section of the root composer.json`.
I do want to both have recipes, and `symfony/flex` in `require-dev` (as decided here https://github.com/symfony/skeleton/issues/24).

So I would like to run e.g. `composer req weblink` (on `dev` env only of course), but it's not working. It would work if `symfony/flex` was `prod` dependency (in `require` section).

**Comment by xabbuh** [comment]
The move of Flex to `require-dev` in the skeleton was reverted in #26.

**Comment by jlekowski** [comment]
Thanks @xabbuh, I will move flex dependecy into `require` then.

**Comment by garak** [comment]
I'm not sure to understand the reason for moving back flex to `require` section, nor the reason to disabling flex when in `require-dev`

**Comment by nicolas-grekas** [comment]
The reason is that flex does important work on composer install also, related to executing scripts. That's why it must be in "require", so that those scripts run also when you deploy your prod. Then, disabling flex in require-dev makes it more evident that it should be moved to require.

**Comment by garak** [comment]
Right, but what if I don't need to run auto-scripts? That's a practical case, I'm running other deploy commands and I can perfectly deploy a Symfony application without flex. But I don't want to give up the opportunity to run flex during development, so why you want to force me to use flex in production?

**Comment by nicolas-grekas** [comment]
I'm just giving you the rationale so you can challenge it.

**Comment by garak** [comment]
And I absolutely appreciate that. But I'm wondering if there can be room for an improvement, giving the option to developers to allow flex in `require-dev` instead of forcing them (of course, defaulting to `require`, as explained). If so, I can open an issue.

**Comment by nicolas-grekas** [comment]
See #415

**Comment by jlekowski** [comment]
Feels like I opened Pandora's box :)

I agree with @garak regarding this - keeping flex in `require-dev` would be nice option to have.

Thanks @nicolas-grekas for taking your time on the issue and this PR https://github.com/symfony/flex/pull/415

**Comment by kosztadani** [comment]
This change still breaks our workflow.

We have a 'developer' composer project that lists  'symfony/flex' in its 'require' section. (This project contains some useful tools, e.g. to set up a database.) We have several small libraries and we used to list this 'developer' project in their 'require-dev' section. This way, when we created such a new library, we could run `composer install` and they would become a full flex project that is convenient to work with. Now this no longer works because they only _indirectly_ depend on flex.

As a workaround, we could add `"symfony/flex": "*"` to their 'require-dev' section. Is this the way to go forward or do you intend to change the behaviour of this?

**Comment by nicolas-grekas** [comment]
@kosztadani see #415, you should upgrade to v1.1.1

**Comment by kosztadani** [comment]
@nicolas-grekas I saw that and I took it into account in the above post. When "symfony/flex" is listed neither in 'require', nor 'require-dev', the indirect dependency doesn't install our small libraries as flex packages.

If I make the 'developer' project dependant on exactly version 1.0.78 of symfony/flex, then simply by listing the 'developer' in a 'require-dev' of a small library, composer will install this as a flex project. Now it now longer does this, unless I list symfony/flex in the small library's 'require-dev' as well.
